lib/vehicle.rb

```diff
@@ -22,4 +26,21 @@ def antique?
   def electric_vehicle?
     @engine == :ev
   end
+
+  def set_registration_date
+    @registration_date = Date.today
+  end
+
+  def assign_plate_type
+    if @registration_date
+
+     @plate_type = if antique?
+        :antique
+      elsif electric_vehicle?
+        :ev
+      else
+        :regular
+      end
+    end
+  end
 end
```

Comment:
Looping back around to talk about these methods ( I had commented on them earlier). These methods will only update the plate type and registration date once they were called. The functionality we were looking for was that the plate type and the registration date were set AS SOON as the car was registered.